### PR TITLE
build: relax the version constraint for Protobuf

### PR DESCRIPTION
**Pull request by tchaikov**
as higher versions of Protobuf does not claim to be compatible with
v2.5.0, so we should not specify a version number which is acceptable
by Seastar as a parameter of `find_package()`. and instead, we just
check the version number after calling `find_package()` in config mode.
if we find the package using config mode, we search it again, for
printing out the version number to the stdout, otherwise we search
using its C++ API version.

Refs ##2113

### CODE REVIEW

File: cmake/SeastarDependencies.cmake [hunk 1]
```diff
@@ -159,11 +159,11 @@ macro (seastar_find_dependencies)
   # ProtobufConfig.cmake provided by protobuf defines this linkage. so we try
   # the CMake package configuration file first, and fall back to CMake's
   # FindProtobuf module.
-  find_package (Protobuf 2.5.0
+  find_package (Protobuf 22.0...25.3
```
Comment (avikivity) [inline cmake/SeastarDependencies.cmake:162]: I don't understand how 2.5.0 worked before and now it's 22.0..22.3. I understood there are two versioning systems, but you don't change versioning systems here.

Comment (avikivity) [reply cmake/SeastarDependencies.cmake:162]: Also, does 26 not work?

**Comment by mykaul** [comment]
Is it somehow aligned with the version Prometheus uses (CC @amnonh ) ?

File: cmake/SeastarDependencies.cmake (hunk 1 above)
Comment (tchaikov) [reply cmake/SeastarDependencies.cmake:162]: @avikivity 

> I don't understand how 2.5.0 worked before and now it's 22.0..22.3. I understood there are two versioning systems, but you don't change versioning systems here.

22.0 is the first version which uses abseil. so i wanted to make this explicit. and i didn't want to mix two versioning systems in a single condition, like `find_package (Protobuf 2.5...25.3)` even if it would yield the expected outcome. but it's just weird..

> Also, does 26 not work?

yeah, i asked myself when writing these numbers. but i wanted to avoid adding versions in future and wanted to avoid pretending that we support versions we haven't even compiled with. that's why it looks this way. but it didn't pan out. 

anyway, i am pushing an updated version.

**Comment by tchaikov** [comment]
v2:

- do not enforce version constraints when finding protobuf in config mode, as it turns out protobuf's cmake configuration does not claim the compatibility with other versions part from its exact version.
- check the minimal required version of `2.5.0` instead of the minimal version which introduced abseil libraries linkage. simpler and more consistent this way.

**Comment by tchaikov** [comment]
> Is it somehow aligned with the version Prometheus uses (CC @amnonh ) ?

@mykaul hi Yaniv, this change is intended to address the FTBFS on a very bleeding edge distro which

1. packages a very new protobuf
2. packages protobuf's CMake configuration file.

the metrics' on-wire protocol is defined by https://buf.build/gogo/protobuf/file/4df00b267f944190a229ce3695781e99:gogoproto/gogo.proto and https://github.com/prometheus/prometheus/blob/main/prompb/io/prometheus/client/metrics.proto. when it comes to the C++ language binding. i think protobuf compiler *should* be backward compatible in this regards.

**Comment by tchaikov** [comment]
per @mickey298 's test at https://github.com/scylladb/seastar/issues/2113#issuecomment-1981213761 . the change does address the FTBFS.

**Comment by tchaikov** [comment]
@scylladb/seastar-maint hello maintainers, could you review this change?

File: cmake/SeastarDependencies.cmake (hunk 1 above)
Comment (avikivity) [reply cmake/SeastarDependencies.cmake]: What I didn't understand is the version jump from 2 to 22. Was the numbering changed? Or do the version numbers refer to different things?

Comment (tchaikov) [reply cmake/SeastarDependencies.cmake]: please note, in the latest revision i stopped using 22 as the version number. and sorry, i failed to explain my reasoning on the previous revision in an understandable way. let me rephrase it.

more like it refer to a different thing. 

actually i didn't bumped the version from 2 to 22. it was because, the CMake configuration file shipped by Protobuf uses the library's version, i.e., 22 and 23..., while the `FindProtobuf` module uses the language binding's version, like 2.5, 3.12, 3.21, 4.22, ....  .  the cutting edge distros are more likely to use the CMake configuration file shipped by Protobuf, so i intended to use the library's version when finding the package using the config mode, and i need to pick a library version . instead of trying to map 2.5 to "protobuf v5.0" which does not exst at all. i guess the reason is that it predates the version when protobuf developers decided to using this new versioning approach. so, i used v22, which is the first [version](https://github.com/protocolbuffers/protobuf/releases/tag/v22.0) which uses abseil. so that i can assure that the change works.

Comment (tchaikov) [reply cmake/SeastarDependencies.cmake]: @avikivity ^

Comment (avikivity) [reply cmake/SeastarDependencies.cmake]: ah, thanks.
